packages/wrangler/e2e/multiworker-dev.test.ts

```diff
@@ -0,0 +1,321 @@
+						name = "${workerName}"
+						main = "src/index.ts"
+						compatibility_date = "2023-01-01"
+
+						[[services]]
+						binding = "BEE"
+						service = '${workerName2}'
+				`,
+			});
+		});
+		it("can fetch b", async () => {
+			const worker = helper.runLongLived(`wrangler dev`, { cwd: b });
+
+			const { url } = await worker.waitForReady(5_000);
+
+			await expect(fetch(url).then((r) => r.text())).resolves.toBe(
+				"hello world"
+			);
+		});
+
+		it("can fetch b through a", async () => {
+			const workerA = helper.runLongLived(
+				`wrangler dev -c wrangler.toml -c ${b}/wrangler.toml`,
+				{ cwd: a }
+			);
+			const { url } = await workerA.waitForReady(5_000);
+
+			await vi.waitFor(
+				async () => await expect(fetchText(url)).resolves.toBe("hello world"),
+				{ interval: 1000, timeout: 10_000 }
+			);
+		});
+
+		it("shows error on startup with non-existent service", async () => {
+			await baseSeed(a, {
+				"wrangler.toml": dedent`
+						name = "${workerName}"
+						main = "src/index.ts"
+						compatibility_date = "2023-01-01"
+
+						[[services]]
+						binding = "BEE"
+						service = '${randomUUID()}'
+				`,
+			});
+
+			const workerA = helper.runLongLived(
+				`wrangler dev -c wrangler.toml -c ${b}/wrangler.toml`,
+				{ cwd: a }
+			);
+			await workerA.readUntil(/no such service is defined/);
+
+			expect(await workerA.exitCode).toBe(1);
+		});
+
+		it(`leaves no orphaned workerd processes after shutdown`, async () => {
+			const beginProcesses = getStartedWorkerdProcesses(helper.tmpPath);
+
+			const workerA = helper.runLongLived(
+				`wrangler dev -c wrangler.toml -c ${b}/wrangler.toml`,
+				{ cwd: a }
+			);
+			const { url } = await workerA.waitForReady(5_000);
+
+			await vi.waitFor(
+				async () => await expect(fetchText(url)).resolves.toBe("hello world"),
+				{ interval: 1000, timeout: 10_000 }
+			);
+
+			await workerA.signal("SIGINT");
+
+			const exitCode = await workerA.exitCode;
+
+			expect(exitCode).toBe(0);
+
+			const endProcesses = getStartedWorkerdProcesses(helper.tmpPath);
+
+			expect(beginProcesses.length).toBe(0);
+			expect(endProcesses.length).toBe(0);
+		});
+	});
+
+	describe("service workers", () => {
+		beforeEach(async () => {
+			await baseSeed(a, {
+				"wrangler.toml": dedent`
+					name = "${workerName}"
+					main = "src/index.ts"
+					compatibility_date = "2023-01-01"
+
+                    [[services]]
+					binding = "CEE"
+					service = '${workerName3}'
+			`,
+			});
+		});
+
+		it("can fetch service worker c through a", async () => {
+			const workerA = helper.runLongLived(
+				`wrangler dev -c wrangler.toml -c ${c}/wrangler.toml`,
+				{ cwd: a }
+			);
+			const { url } = await workerA.waitForReady(5_000);
+
+			await vi.waitFor(
+				async () =>
+					await expect(fetchText(`${url}/service`)).resolves.toBe(
+						"Hello from service worker"
+					),
+				{ interval: 1000, timeout: 10_000 }
+			);
+		});
+	});
+
+	describe("durable objects", () => {
```

Comment:
would be nice to get some tests for RPC too (to DO and regular worker), since that's explicitly called out in the changeset. And named entrypoints too maybe. don't want to block on this, but would be good to get confirmation its been tested manually + followup with it?